resources/analysisTools/qcPipeline/flags_isizes_PEaberrations.pl

```diff
@@ -180,62 +171,55 @@
 		}
 	}
 }
-close IN;
+close $samFh;
 # extended flagstats
-print FL "total alignments\t$all\nnon-duplicate, non-secondary, non-supplementary reads\t$uniq\nsuch with mapping quality >=$minmapq\t$minmapuniq\nsuch on regarded chromosomes\t$onchr\nsuch with both reads on regarded chromosomes\t$both\nsuch mapping on different chromosomes\t$aberrant\nproper pairs read 1\t$ctr\n";
-close FL;
+print $flagsFh "total alignments\t$all\nnon-duplicate, non-secondary, non-supplementary reads\t$uniq\nsuch with mapping quality >=$minmapq\t$minmapuniq\nsuch on regarded chromosomes\t$onchr\nsuch with both reads on regarded chromosomes\t$both\nsuch mapping on different chromosomes\t$aberrant\nproper pairs read 1\t$ctr\n";
+close $flagsFh;
 
+if ($ctr == 0) {
+	print STDERR "No properly paired reads found! Is this single end data? Is the input file truncated?\n";
+}
 
 # in case we have single end reads, there will be no counted ones for isizes ($ctr < 1)
 # and PE aberrations ($aberrant < 1) => fill the files with placeholder "NA"
 
-if ($aberrant < 1)
-{
-	print STDERR "no aberrant paired reads found, single end reads?\n";
-	print PE "NA\n";
-}
-else
-{
-	$average = $sum/$ctr;
+my $percentage = "NA";
+if ($aberrant < 1) {
+	print STDERR "No aberrant paired reads found, single end reads?\n";
+} else {
+	$average = "NA";
+	if ($ctr > 0) {
+		$average = $sum / $ctr;
+	}
 	print STDERR "insert sizes: $pp_strange proper pairs with insert size > $maxisize; $ctr values in range. minimum: $min; maximum: $max; average: $average\n";
 	print STDERR "from $all reads, $aberrant are paired end aberrations\n";
-	my $percentage = sprintf ("%2.2f\n", ($aberrant/$both*100));
-	print PE $percentage;
+	if ($both > 0) {
+		$percentage = sprintf("%2.2f", ($aberrant / $both * 100));
+	}
```

Comment:
Would it make sense to output something for else as well?

Reply:
Note this is sprintf(), not printf(). For else, the default value (during initialization) 'NA' is returned. See print-statement below.